float32 vel_1
float32 vel_2
float32 vel_3
float32 vel_4